packages/hoppscotch-common/src/components/collections/SaveRequest.vue

```diff
@@ -126,12 +126,13 @@ const emit = defineEmits<{
 }>()
 
 const gqlRequestName = useGQLRequestName()
-const requestName = computedWithControl(
-  () => [currentActiveTab.value, gqlRequestName.value],
-  () =>
-    props.mode === "rest"
-      ? currentActiveTab.value.document.request.name
-      : gqlRequestName.value
+const restRequestName = computedWithControl(
+  () => currentActiveTab.value,
+  () => currentActiveTab.value.document.request.name
+)
+
+const requestName = ref(
```

Comment:
Here since the ref is initialized when the component is rendered first. it won't be updated if there is an update to the currentActiveTab.value, for eg: try renaming the request, and see the value in the save request model.

you can add a watcher on the restRequestName that will update the ref incase the name changes from the tabs side. 

```javascript
watch(restRequestName, (newValue) => {
  requestName.value = newValue
})
```